e2etest/l2tests/assignment.go

```diff
@@ -678,4 +698,271 @@ var _ = ginkgo.Describe("IP Assignment", func() {
+				Expect(err).NotTo(HaveOccurred())
+				return len(svc1.Status.LoadBalancer.Ingress)
+			}, 5*time.Minute, 1*time.Second).Should(Equal(0))
+
+			ginkgo.By("Updating svc1 to have PreferDualStack family policy")
+			err = updateServiceIPFamilyPolicy(cs, svc1.Namespace, svc1.Name, ptr.To(v1.IPFamilyPolicyPreferDualStack))
+			Expect(err).NotTo(HaveOccurred())
+			ginkgo.By("Validating that 1 IP is assigned to the service")
+			Eventually(func() int {
+				svc1, err = cs.CoreV1().Services(svc1.Namespace).Get(context.Background(), svc1.Name, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				return len(svc1.Status.LoadBalancer.Ingress)
+			}, 5*time.Minute, 1*time.Second).Should(Equal(1))
+			ginkgo.By("validate LoadBalancer IP is allocated from pool1")
+			err = config.ValidateIPInRange([]metallbv1beta1.IPAddressPool{pool1}, jigservice.GetIngressPoint(
+				&svc1.Status.LoadBalancer.Ingress[0]))
+			Expect(err).NotTo(HaveOccurred())
+		})
+		ginkgo.It("Svc assigned from single-stack pool should not jump to dual-stack pool when it's available", func() {
+			resources := config.Resources{
+				Pools: []metallbv1beta1.IPAddressPool{poolv4},
+			}
+
+			err := ConfigUpdater.Update(resources)
+			Expect(err).NotTo(HaveOccurred())
+
+			ginkgo.By("Creating the service with PreferDualStack policy")
+			svc1, _ := service.CreateWithBackend(cs, testNamespace, "svc-test-ns-pool-1", func(svc *v1.Service) {
+				svc.Spec.IPFamilyPolicy = ptr.To(v1.IPFamilyPolicyPreferDualStack)
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			ginkgo.By("Validating that 1 IP is assigned to the service")
+			Eventually(func() int {
+				svc1, err = cs.CoreV1().Services(svc1.Namespace).Get(context.Background(), svc1.Name, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				return len(svc1.Status.LoadBalancer.Ingress)
+			}, 5*time.Minute, 1*time.Second).Should(Equal(1))
+			ginkgo.By("validate LoadBalancer IP is allocated from poolv4")
+			err = config.ValidateIPInRange([]metallbv1beta1.IPAddressPool{poolv4}, jigservice.GetIngressPoint(
+				&svc1.Status.LoadBalancer.Ingress[0]))
+			Expect(err).NotTo(HaveOccurred())
+			firstIP := svc1.Status.LoadBalancer.Ingress[0].String()
+			ginkgo.By("Updating pools to include dual-stack pool")
+			resources = config.Resources{
+				Pools: []metallbv1beta1.IPAddressPool{poolv4, poolDual},
+			}
+			err = ConfigUpdater.Update(resources)
+			Expect(err).NotTo(HaveOccurred())
+			// Sleep to let the svc jump if it wants
+			time.Sleep(2 * time.Minute)
+			ginkgo.By("Validating that the originally assigned IP hasn't changed")
+			Eventually(func() bool {
+				svc1, err = cs.CoreV1().Services(svc1.Namespace).Get(context.Background(), svc1.Name, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				return svc1.Status.LoadBalancer.Ingress[0].String() == firstIP
+			}, 1*time.Minute, 1*time.Second).Should(BeTrue())
```

Comment:
don't use sleeps, this should be a `Consistently`

Reply:
Thank you! I've updated it.